pre-commit.sh

```diff
@@ -5,14 +5,23 @@ exec 1>&2
 # No tests to update gh-pages
 BRANCH=$(git symbolic-ref --short HEAD 2>/dev/null)
 if [ "$BRANCH" = "gh-pages" ] ||
+   [ "$BRANCH" = "gh-issues" ] ||
    [ -e .git/MERGE_HEAD ]; then
      exit 0
 fi
 
-git stash save -k -q
+STASHED=0
+if test `git status --porcelain | grep '^[A-Z]' | wc -l` -gt 0; then
```

Comment:
No, because you can have staged a valid version, and since made a breaking change that isn't part of the commit.  I don't think that works if you just ignore files with changes -- what you want is to check that whatever's currently staged for commit will build successfully.

I replaced the `test` with `[]`; hopefully that addresses the style concern.